### PR TITLE
Add RTL support to Stack

### PR DESCRIPTION
**Pull request by abarth**
Notice that the default alignment for Stack needs to know the
directionality.

### CODE REVIEW

**Comment by Hixie** [comment]
It's a bit weird that the alignment of non-positioned children is resolved by the render object, but the alignment of positioned children is resolved by the widgets layer.

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/test/widgets/app_overrides_test.dart [hunk 1]
```diff
@@ -26,12 +26,17 @@ class TestRoute<T> extends PageRoute<T> {
...
 
 Future<Null> pumpApp(WidgetTester tester) async {
-  await tester.pumpWidget(new WidgetsApp(
-    color: const Color(0xFF333333),
-    onGenerateRoute: (RouteSettings settings) {
-      return new TestRoute<Null>(settings: settings, child: new Container());
-    },
-  ));
+  await tester.pumpWidget(
+    new Directionality(
```
Comment (Hixie) [inline packages/flutter/test/widgets/app_overrides_test.dart]: It turns out that this hid some real bugs. WidgetsApp is the widget that's supposed to introduce the Directionality. :-)

I'll post a fix.

Comment (abarth) [reply packages/flutter/test/widgets/app_overrides_test.dart]: Thanks :)

**Pull request merged**